static/docs/commands-reference/fetch.md

```diff
@@ -1,62 +1,243 @@
+available for linking (or copying) into the workspace. (Refer to
+[dvc config cache.type](https://dvc.org/doc/commands-reference/config#cache).)
+
+Note, `dvc fetch` is performed automatically by `dvc pull` when the target stage
+files are not already in the local cache. Specifically,
+`dvc pull` = `dvc fetch` + `dev checkout`:
 
 ```
+Controlled files             Commands
+---------------- ---------------------------------
+
+ remote storage
+       +
+       |         +------------+ 
+       | - - - - | dvc fetch  | ++
+       v         +------------+   +   +----------+
+  local cache                      ++ | dvc pull |
+       +         +------------+   +   +----------+
+       | - - - - |dvc checkout| ++
+       |         +------------+
+       v
+   workspace
+```
+
+## Options
+
+- `-r REMOTE`, `--remote REMOTE` - name of the
+  [remote storage](https://dvc.org/doc/commands-reference/remote#description)
+  to fetch from (see `dvc remote list`). If not specified, the default 
+  remote is used (see `dvc config core.remote`). The argument `REMOTE` is a
+  remote name defined using the `dvc remote` command.
+
+- `-d`, `--with-deps` - fetch cache by tracking dependencies to the named
+  target stages. This option only has effect when one or more `targets` are
+  specified. By traversing the dependencies, DVC searches backward through the
+  pipeline from the named target(s). This means DVC will not fetch files needed
+  later in the pipeline than the named target(s).
+
+- `-R`, `--recursive` - this option tells DVC that `targets` are
+  directories (not DVC files), and to traverse them recursively. All DVC 
+  files found will be read in order to determine the set of data files to fetch.
+
+- `-j JOBS`, `--jobs JOBS` - number of threads to run simultaneously to 
+  handle the downloading of files from the remote. Using more jobs may 
+  improve the total download speed if a combination of small and large files 
+  are being fetched.
+  The default value is `4 * cpu_count()`. For SSH remotes default is 4.
+  > Note, this applies only to local remotes and for SSH remotes. (See `dvc 
+  remote` for more information about  remotes.)
+
+- `-a`, `--all-branches` - fetch cache for all branches, not just the active
+  one. This means that you'll the files needed to reproduce different
+  versions of a DVC file ([experiments](/doc/get-started/experiments)), not
+  just the current one.
+
+- `-T`, `--all-tags` - fetch cache for all tags. Similar to `-a` above
+
+- `--show-checksums` - show checksums instead of file names when printing the
+  download progress.
+
+- `-v`, `--verbose` - displays detailed tracing information.
+
+- `-q`, `--quiet` - do not write anything to standard output.
+
+- `-h`, `--help` - prints the usage/help message, and exit.
 
 ## Examples
 
-Fetch all files used in the current Git branch:
+Let's add a couple remotes to the project first, so DVC can control our files
```

Comment:
Please, let's adopt the framework from the latest `dvc checkout` version I made a few days ago and mentioned before. This is important because we are trying to make all examples runnable. So, I've prepared a Github repo with tags, pre-built S3 remote, etc. So that users can play with for real.

We could also use second example like in `dvc repro` - something artificial. Like a really quick way to show and illustrate how different options work.

Reply:
I will... Still working on improving examples. Definitely the most tricky part

Reply:
Done.

Please note I left the `## Examples: Specific stages` and other sub-headers similar to the `dvc pull` and `dvc push` refs as requested earlier but I noticed those aren't used in `dvc checkout`.

Reply:
I'm also thinking about adding the `## Examples: Specific remotes` and `## Example: Specific Git tags` examples but the section is already pretty long. Do you think they are desirable? Thanks

Reply:
I think we have more than enough for this iteration! :)

Reply:
Agreed